src/main/java/uwu/command/AddCommand.java

```diff
@@ -0,0 +1,109 @@
+package uwu.command;
+
+import uwu.exception.EmptyInputException;
+import uwu.exception.IncorrectFormatException;
+import uwu.exception.InvalidDateException;
+import uwu.exception.UwuException;
+
+import uwu.Storage;
+
+import uwu.task.Deadline;
+import uwu.task.Event;
+import uwu.task.Task;
+import uwu.task.TaskList;
+import uwu.task.ToDos;
+
+import uwu.Ui;
+
+/**
+ * Adds a task to the task list.
+ */
+public class AddCommand extends Command {
+    /** The description of the task. */
+    String description;
+
+    /** The type of task to be added. */
+    String taskType;
+
+    /** The user command parsed into AddCommand. */
+    String userCommand;
+
+    /**
+     * Constructor for an AddCommand object.
+     *
+     * @param userCommand The command the user typed.
+     */
+    public AddCommand(String userCommand) {
+        this.userCommand = userCommand;
+        String[] taskData = userCommand.split(" ", 2);
+        this.taskType = taskData[0];
+        this.description = taskData[1];
+    }
+
+    /**
+     * Executes the AddCommand which adds a task of taskType into a stored task list.
+     *
+     * @param tasks The list where tasks are added to.
+     * @param ui The ui to print out UwuBot's response.
+     * @param storage The task list that is stored in the user's hard disk.
+     * @throws UwuException If task description is empty;
+     *                      If task does not contain keyword;
+     *                      If date is empty for deadline and event tasks.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws UwuException {
+        if (userCommand.replaceFirst(taskType, "").isBlank()) {
+            throw new EmptyInputException("\tyour task description is empty TT\n\t"
+                    + "feed me a task description to get started! <:");
+        }
+
+        switch (taskType) {
+        case "todo":
+                ToDos todo = new ToDos(description);
+                tasks.add(todo);
+                storage.save(tasks.taskListToStorageString());
+                ui.addTask(todo, tasks.size());
+                break;
+        case "deadline":
```

Comment:
In the case whenever there is a case statement without a break statement, I believe that there should be an explicit //Fallthrough comment that should be included